### DISCUSSION TITLE
Question: Is it possible to embed a perspective viewer in a merged row as hierarchy child

### DISCUSSION THREAD

**Discussion by khuntia** [perspective-dev/perspective#2733] [Q&A]
Hi Perspective Team

I am evaluating perspective to see if this will work for our usecase. I am impressed with the responsiveness and speed !

I have a further noob question. My usecase is that of highly scalable table but with hierarchy. Ie like a list table or tree-table.

I see this is possible in perspective as we can collapse/expand rows to expose more rows. What I want to understand is when I expand a row, its children which get inserted, i want each child to be another completely independent perspective viewer with a different table (Obviously this row will then have a lot bigger dimension of width and height). In effect it opens up another table itself (with its own headers and data). And when collapsed that table gets un-rendered/destroyed.

I was thinking if we merge the cells and make it one cell and then the content of the cell being another new viewer. Will this work? I am sorry I am not well versed on perspective and the documentation doesnt make this clear yet. 

Thanks
Abhi

**Comment by aszenz** [comment]
You can build a tree-table with the combination of group by + unique aggregation

Also explore global filter on perspective workspace for an alternative solution
